packages/pages-shared/src/asset-server/rulesEngine.ts

```diff
@@ -67,11 +83,17 @@ export const generateRulesMatcher = <T>(
 	][];
 
 	return ({ request }: { request: Request }) => {
-		const { pathname, host } = new URL(request.url);
+		const { pathname, host, search } = new URL(request.url);
+		const sortedSearch =
+			search && search.length
+				? "?" + search.slice(1).split("&").sort().join("&")
+				: search;
 
 		return compiledRules
 			.map(([{ crossHost, regExp }, match]) => {
-				const test = crossHost ? `https://${host}${pathname}` : pathname;
+				const test = crossHost
+					? `https://${host}${pathname}${sortedSearch}`
+					: `${pathname}${sortedSearch}`;
```

Comment:
```suggestion
					: `${pathname}${sortedSearch.length > 0 ? sortedSearch : ""}`;
```

Reply:
This is already an empty string, that's unnecessary